enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/ChoreoFaultAnalyticsProvider.java

```diff
@@ -142,16 +145,20 @@ public Application getApplication() {
     @Override
     public Operation getOperation() {
         // This could be null if  OPTIONS request comes
-        if (requestContext.getMatchedResourcePath() != null) {
+        if (requestContext.getMatchedResourcePaths() != null) {
             Operation operation = new Operation();
             if (isWebsocketUpgradeRequest) {
                 operation.setApiMethod(WebSocketFrameRequest.MessageDirection.HANDSHAKE.name());
                 operation.setApiResourceTemplate(AnalyticsConstants.WEBSOCKET_HANDSHAKE_RESOURCE_PREFIX +
-                        requestContext.getMatchedResourcePath().getPath());
+                        requestContext.getMatchedResourcePaths().get(0).getPath());
                 return operation;
             }
-            operation.setApiMethod(requestContext.getMatchedResourcePath().getMethod().name());
-            operation.setApiResourceTemplate(requestContext.getMatchedResourcePath().getPath());
+            operation.setApiMethod(requestContext.getMatchedResourcePaths().get(0).getMethod().name());
+            ArrayList<String> resourceTemplate = new ArrayList<>();
+            for (ResourceConfig resourceConfig: requestContext.getMatchedResourcePaths()) {
+                resourceTemplate.add(resourceConfig.getPath());
+            }
+            operation.setApiResourceTemplate(String.join(",", resourceTemplate));
```

Comment:
Do we have multiple resource templates here

Reply:
yes for gql apis, multiple operation names would be there.

Reply:
Since the choreo analytics manages these resources properly at their end, we are good to keep it in this format. (Counters are incremented separately)